inc/class-wpseo-replace-vars.php

```diff
@@ -111,8 +111,12 @@ public static function register_replacement( $var, $replace_function, $type = 'a
 			}
 			elseif ( ! method_exists( __CLASS__, 'retrieve_' . $var ) ) {
 				if ( ! isset( self::$external_replacements[ $var ] ) ) {
+					if ( empty( $var ) ) {
```

Comment:
Maybe we also should check if this is not a string as well:
`if( empty( $var ) || ! is_string( $var )`